// @memberstack/dom/lib/utils/cookies.mjs
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};

// src/utils/cookies.ts
import Cookie from "js-cookie";
var memberAuthTokenName = "_ms-mid";
function isLocalStorageAvailable() {
  try {
    localStorage.setItem("test", "test");
    localStorage.removeItem("test");
    return true;
  } catch (e) {
    return false;
  }
}
var localStorageAvailable = isLocalStorageAvailable();
var useCookies = false;
var setCookieOnRootDomain = false;
var apiDomain;
var setUseCookies = (setCookieOnRoot) => {
  useCookies = true;
  if (setCookieOnRoot)
    setCookieOnRootDomain = true;
};
var setApiDomain = (domain) => {
  apiDomain = domain;
};
var extractDomainFromHostname = (hostname) => {
  const hostnameParts = hostname.split(".");
  const isCountryCodeTLD = hostnameParts.length >= 3 && hostnameParts[hostnameParts.length - 2].length <= 3;
  if (isCountryCodeTLD) {
    return hostnameParts.slice(-3).join(".");
  } else {
    return hostnameParts.slice(-2).join(".");
  }
};
var useCookiesValid = () => {
  if (!useCookies)
    return false;
  if (setCookieOnRootDomain) {
    const apiDomainHost = extractDomainFromHostname(apiDomain);
    const currentHost = extractDomainFromHostname(window.location.hostname);
    return apiDomainHost === currentHost;
  }
  return true;
};
var setMemberToken = (token, expires) => {
  if (localStorageAvailable && useCookiesValid() === false) {
    localStorage.setItem(memberAuthTokenName, token);
  } else {
    const domain = extractDomainFromHostname(window.location.hostname);
    Cookie.set(memberAuthTokenName, token, __spreadValues({
      expires: expires && Number(expires) || 14,
      sameSite: "strict"
    }, setCookieOnRootDomain && { domain }));
  }
};
var removeMemberToken = () => {
  if (localStorageAvailable) {
    localStorage.removeItem(memberAuthTokenName);
  }
  const domain = extractDomainFromHostname(window.location.hostname);
  Cookie.set(memberAuthTokenName, "", __spreadValues({
    expires: -1,
    sameSite: "strict"
  }, setCookieOnRootDomain && { domain }));
  Cookie.set(memberAuthTokenName, "", {
    expires: -1,
    sameSite: "strict"
  });
};
var getMemberToken = () => {
  if (localStorageAvailable) {
    const memAuthToken = localStorage.getItem(memberAuthTokenName);
    if (memAuthToken)
      return memAuthToken;
  }
  return Cookie.get(memberAuthTokenName);
};
var setMemberTokenIfAvailable = () => {
  if (typeof window !== "undefined") {
    const urlParams = new URLSearchParams(window.location.search);
    const memberToken = urlParams.get("memberToken");
    if (memberToken)
      setMemberToken(memberToken);
  }
};
var getSessionId = () => {
  if (typeof window === "undefined")
    return null;
  const sessionId = window.localStorage.getItem("ms_session_id");
  if (!sessionId || sessionId.length > 26) {
    window.localStorage.removeItem("ms_session_id");
    return null;
  }
  return sessionId;
};
var setSessionId = (sessionId) => {
  if (typeof window !== "undefined") {
    window.localStorage.setItem("ms_session_id", sessionId);
  } else {
    console.warn("setSessionId: window is not defined");
  }
};
var removeSessionId = () => {
  if (typeof window !== "undefined") {
    window.localStorage.removeItem("ms_session_id");
  } else {
    console.warn("removeSessionId: window is not defined");
  }
};
export {
  extractDomainFromHostname,
  getMemberToken,
  getSessionId,
  removeMemberToken,
  removeSessionId,
  setApiDomain,
  setMemberToken,
  setMemberTokenIfAvailable,
  setSessionId,
  setUseCookies,
  useCookiesValid
};
