// @memberstack/dom/lib/auth/index.global.js
var MemberstackDOM = (() => {
  var __defProp = Object.defineProperty;
  var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
  var __getOwnPropNames = Object.getOwnPropertyNames;
  var __getOwnPropSymbols = Object.getOwnPropertySymbols;
  var __hasOwnProp = Object.prototype.hasOwnProperty;
  var __propIsEnum = Object.prototype.propertyIsEnumerable;
  var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
  var __spreadValues = (a, b) => {
    for (var prop in b || (b = {}))
      if (__hasOwnProp.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    if (__getOwnPropSymbols)
      for (var prop of __getOwnPropSymbols(b)) {
        if (__propIsEnum.call(b, prop))
          __defNormalProp(a, prop, b[prop]);
      }
    return a;
  };
  var __export = (target, all) => {
    for (var name in all)
      __defProp(target, name, { get: all[name], enumerable: true });
  };
  var __copyProps = (to, from, except, desc) => {
    if (from && typeof from === "object" || typeof from === "function") {
      for (let key of __getOwnPropNames(from))
        if (!__hasOwnProp.call(to, key) && key !== except)
          __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
    }
    return to;
  };
  var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

  // src/auth/index.ts
  var auth_exports = {};
  __export(auth_exports, {
    getPersistedMember: () => getPersistedMember,
    onAuthChange: () => onAuthChange,
    setPersistedMember: () => setPersistedMember,
    unsetPersistedMember: () => unsetPersistedMember,
    updatePersistedMember: () => updatePersistedMember
  });

  // ../../node_modules/js-cookie/dist/js.cookie.mjs
  function assign(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        target[key] = source[key];
      }
    }
    return target;
  }
  var defaultConverter = {
    read: function(value) {
      if (value[0] === '"') {
        value = value.slice(1, -1);
      }
      return value.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent);
    },
    write: function(value) {
      return encodeURIComponent(value).replace(
        /%(2[346BF]|3[AC-F]|40|5[BDE]|60|7[BCD])/g,
        decodeURIComponent
      );
    }
  };
  function init(converter, defaultAttributes) {
    function set(name, value, attributes) {
      if (typeof document === "undefined") {
        return;
      }
      attributes = assign({}, defaultAttributes, attributes);
      if (typeof attributes.expires === "number") {
        attributes.expires = new Date(Date.now() + attributes.expires * 864e5);
      }
      if (attributes.expires) {
        attributes.expires = attributes.expires.toUTCString();
      }
      name = encodeURIComponent(name).replace(/%(2[346B]|5E|60|7C)/g, decodeURIComponent).replace(/[()]/g, escape);
      var stringifiedAttributes = "";
      for (var attributeName in attributes) {
        if (!attributes[attributeName]) {
          continue;
        }
        stringifiedAttributes += "; " + attributeName;
        if (attributes[attributeName] === true) {
          continue;
        }
        stringifiedAttributes += "=" + attributes[attributeName].split(";")[0];
      }
      return document.cookie = name + "=" + converter.write(value, name) + stringifiedAttributes;
    }
    function get(name) {
      if (typeof document === "undefined" || arguments.length && !name) {
        return;
      }
      var cookies = document.cookie ? document.cookie.split("; ") : [];
      var jar = {};
      for (var i = 0; i < cookies.length; i++) {
        var parts = cookies[i].split("=");
        var value = parts.slice(1).join("=");
        try {
          var found = decodeURIComponent(parts[0]);
          jar[found] = converter.read(value, found);
          if (name === found) {
            break;
          }
        } catch (e) {
        }
      }
      return name ? jar[name] : jar;
    }
    return Object.create(
      {
        set,
        get,
        remove: function(name, attributes) {
          set(
            name,
            "",
            assign({}, attributes, {
              expires: -1
            })
          );
        },
        withAttributes: function(attributes) {
          return init(this.converter, assign({}, this.attributes, attributes));
        },
        withConverter: function(converter2) {
          return init(assign({}, this.converter, converter2), this.attributes);
        }
      },
      {
        attributes: { value: Object.freeze(defaultAttributes) },
        converter: { value: Object.freeze(converter) }
      }
    );
  }
  var api = init(defaultConverter, { path: "/" });

  // src/utils/cookies.ts
  var memberAuthTokenName = "_ms-mid";
  function isLocalStorageAvailable() {
    try {
      localStorage.setItem("test", "test");
      localStorage.removeItem("test");
      return true;
    } catch (e) {
      return false;
    }
  }
  var localStorageAvailable = isLocalStorageAvailable();
  var setCookieOnRootDomain = false;
  var extractDomainFromHostname = (hostname) => {
    const hostnameParts = hostname.split(".");
    const isCountryCodeTLD = hostnameParts.length >= 3 && hostnameParts[hostnameParts.length - 2].length <= 3;
    if (isCountryCodeTLD) {
      return hostnameParts.slice(-3).join(".");
    } else {
      return hostnameParts.slice(-2).join(".");
    }
  };
  var removeMemberToken = () => {
    if (localStorageAvailable) {
      localStorage.removeItem(memberAuthTokenName);
    }
    const domain = extractDomainFromHostname(window.location.hostname);
    api.set(memberAuthTokenName, "", __spreadValues({
      expires: -1,
      sameSite: "strict"
    }, setCookieOnRootDomain && { domain }));
    api.set(memberAuthTokenName, "", {
      expires: -1,
      sameSite: "strict"
    });
  };
  var removeSessionId = () => {
    if (typeof window !== "undefined") {
      window.localStorage.removeItem("ms_session_id");
    } else {
      console.warn("removeSessionId: window is not defined");
    }
  };

  // src/auth/index.ts
  var createObservable = () => {
    const observer = /* @__PURE__ */ new Map();
    return {
      subscribe: (key, fn) => {
        if (typeof fn !== "function")
          return;
        return observer.set(key, fn);
      },
      unsubscribe: (key) => {
        return observer.delete(key);
      },
      notify: (data) => {
        return [...observer.values()].forEach((fn) => fn(data));
      }
    };
  };
  var observable = createObservable();
  var getPersistedMember = () => {
    return JSON.parse(localStorage == null ? void 0 : localStorage.getItem("_ms-mem"));
  };
  var setPersistedMember = (member) => {
    if (member) {
      const data = JSON.stringify(member);
      if (data)
        localStorage == null ? void 0 : localStorage.setItem("_ms-mem", data);
      observable.notify(member);
    }
  };
  var updatePersistedMember = (member) => {
    if (member) {
      const data = JSON.stringify(member);
      if (data)
        localStorage == null ? void 0 : localStorage.setItem("_ms-mem", data);
    } else {
      localStorage == null ? void 0 : localStorage.removeItem("_ms-mem");
      removeMemberToken();
    }
  };
  var unsetPersistedMember = () => {
    localStorage == null ? void 0 : localStorage.removeItem("_ms-mem");
    removeMemberToken();
    removeSessionId();
    observable.notify(null);
  };
  var onAuthChange = (cb) => {
    const id = Math.floor(Math.random() * 1e3);
    observable.subscribe(id, cb);
    return {
      unsubscribe: () => observable.unsubscribe(id)
    };
  };
  return __toCommonJS(auth_exports);
})();
/*! Bundled license information:

js-cookie/dist/js.cookie.mjs:
  (*! js-cookie v3.0.5 | MIT *)
*/
